src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2058 @@
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if input_ids is None:
+            raise ValueError("You have to specify input_ids")
+
+        input_shape = input_ids.size()
+        input_ids = input_ids.view(-1, input_shape[-1])
+
+        hidden_states = self.embeddings(input_ids=input_ids, position_ids=position_ids)
+
+        batch_size, seq_len = input_shape
+
+        attention_mask = self._build_attention_mask(
+            attention_mask, batch_size, seq_len, hidden_states.dtype, hidden_states.device
+        )
+
+        encoder_outputs = self.encoder(
+            inputs_embeds=hidden_states,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        last_hidden_state = encoder_outputs[0]
+        last_hidden_state = self.layernorm(last_hidden_state)
+
+        # casting to torch.int for onnx compatibility: argmax doesn't support int64 inputs with opset 14
+        pooled_output = last_hidden_state[
+            torch.arange(last_hidden_state.shape[0], device=last_hidden_state.device),
+            input_ids.to(dtype=torch.int, device=last_hidden_state.device).argmax(dim=-1),
+        ]
+
+        if not return_dict:
+            return (last_hidden_state, pooled_output) + encoder_outputs[1:]
+
+        return ImageBindTransformerOutput(
+            last_hidden_state=last_hidden_state,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+    @staticmethod
+    def _expand_mask(mask: torch.Tensor, dtype: torch.dtype):
+        # Expand and invert the mask, then fill masked areas
+        return (1.0 - mask[:, None, None, :].to(dtype)).masked_fill(mask[:, None, None, :].to(dtype) == 0, torch.finfo(dtype).min)
+
+    def _build_attention_mask(self, attention_mask, batch_size, seq_len, dtype, device=None):
+        # Build causal mask
+        mask = torch.empty(batch_size, seq_len, seq_len, dtype=dtype, device=device)
+        mask.fill_(torch.finfo(dtype).min)
+        mask.triu_(1)
+        mask = mask.unsqueeze(1)  # expand mask
+
+        # If attention_mask update causal mask
+        if attention_mask is not None:
+            attention_mask = self._expand_mask(attention_mask, dtype)
+            return mask + attention_mask
+        return mask
+
+
+@add_start_docstrings(
+    """The text model from ImageBind without any head or projection on top.""",
+    IMAGEBIND_START_DOCSTRING,
+)
+class ImageBindTextModel(ImageBindPreTrainedModel):
```

Comment:
I don't remember what we said about this class but it's a pity to have it as it wraps around `ImageBindTextTransformer` not super useful!